server/services/v1/query_runner.go

```diff
@@ -341,6 +360,9 @@ func (runner *ReplaceQueryRunner) Run(ctx context.Context, tx transaction.Tx, te
 	if err != nil {
 		return nil, ctx, err
 	}
+	if err = runner.mustBeDocumentsCollection(coll, "replace"); err != nil {
```

Comment:
Can we actually do this in validator?

Reply:
No, collection object is created in the service layer. Validator is only meant for request validation.